### PR TITLE
Profile Pic Backend

### PR DESCRIPTION
**Pull request by bbryanb333**

### CODE REVIEW

File: SH4RE/SH4RE/Account/EditAccountView.swift [hunk 1]
```diff
@@ -17,27 +18,49 @@ struct EditAccountView: View {
...
+        let docRef = Firestore.firestore().collection("User Info").document(getCurrentUserUid())
+        
         dataToChange.removeAll()
         if (name != user.name && name != "") {
-            dataToChange["name"] = name
+            documentUpdate(collectionPath: "User Info", documentID: getCurrentUserUid(), data: ["name": name])
         }
-        if (pfpChanged) {
-            dataToChange["pfp"] = profilePicture
+        else if (pfpChanged) {
```
Comment (WasifButt) [inline SH4RE/SH4RE/Account/EditAccountView.swift]: this and below should if not else if, cause user might want to change multiple things

File: SH4RE/SH4RE/Messages/MessagesInboxView.swift [hunk 2]
```diff
@@ -43,6 +45,7 @@ struct MessagesInboxView: View {
                 }.onAppear(){
                     vm.fetchCurrentUser()
                     vm.fetchRecentMessages()
+                    //feprint(vm.recentMessages)
```
Comment (WasifButt) [inline SH4RE/SH4RE/Messages/MessagesInboxView.swift]: remove this

File: SH4RE/SH4RE/Messages/MessagesInboxView.swift [hunk 3]
```diff
@@ -67,20 +70,23 @@ struct MessagesInboxView: View {
...
-                            
                             self.chatLogViewModel.chatUser = self.chatUser
                             self.chatLogViewModel.fetchMessages()
+                            self.chatLogViewModel.profilePic = profilePicDict[recentMessage.toId] ?? UIImage(named: "ProfilePhotoPlaceholder")!
                             self.shouldNavigateToChatLogView.toggle()
+                            
                         } label: {
                             HStack(spacing: 16) {
-                                Image("ProfilePhotoPlaceholder") // TODO: this will become a profile picture
+                                Image(uiImage: profilePicDict[recentMessage.toId] ?? UIImage(named: "ProfilePhotoPlaceholder")!) // TODO: this will become a profile picture
```
Comment (WasifButt) [inline SH4RE/SH4RE/Messages/MessagesInboxView.swift]: can remove TODO comment

**Review by WasifButt — CHANGES_REQUESTED** [review]

File: SH4RE/SH4RE/Account/EditAccountView.swift (hunk 1 above)
Comment (bbryanb333) [reply SH4RE/SH4RE/Account/EditAccountView.swift]: true

File: SH4RE/SH4RE/Messages/MessagesInboxView.swift (hunk 3 above)
Comment (bbryanb333) [reply SH4RE/SH4RE/Messages/MessagesInboxView.swift]: We must remind ourselves of our past successes

File: SH4RE/SH4RE/Messages/MessagesInboxView.swift (hunk 2 above)
Comment (bbryanb333) [reply SH4RE/SH4RE/Messages/MessagesInboxView.swift]: i kinda liked it tho :(

**Review by WasifButt — DISMISSED** [review]

**Pull request merged**